server.go

```diff
@@ -68,11 +68,14 @@ func (s *Server) Serve(l net.Listener) error {
 	for {
 		conn, err := l.Accept()
 		if err != nil {
+			s.mu.Lock() // sync access to done channel
```

Comment:
channel access should be atomic right? No need for locks.

Reply:
I am trying to hunt down the CI failures. The goal was to to have this loop exit before exposing the accept error.